server/src/utils/getOrInitialiseSolFileEntry.ts

```diff
@@ -10,19 +12,13 @@ import { SolFileEntry } from "../parser/analyzer/SolFileEntry";
  * Uri needs to be decoded and without the "file://" prefix.
  */
 export function getOrInitialiseSolFileEntry(
-  {
```

Comment:
You can always pass the serverState to everything as a giant global, but it does mean it is more work to figure out what state a function is dependent on. If you try and test the function, then suddenly you need an entire `serverState` object or you need to lie and hack the type system.

Reply:
Don't feel the need to change this back. Just explaining why I seem to be doing odd things.

Reply:
I agree, I didn't pay too much attention to that but it makes complete sense. In this case, it's needed because of the ProjectlessProject creation inside `getOrInitialiseSolFileEntry`, which I'd like to remove. It's there for cases where for example, you have a workspace folder open but open a random solidity file from another folder into the current workspace. Then it needs to initialize a SolFileEntry for it and its dependencies. I'll see if I can remove it